src/providers/blocknative-gas-price-provider.ts

```diff
@@ -0,0 +1,90 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import retry from 'async-retry';
+import axios from 'axios';
+
+import { log } from '../util/log';
+
+import { GasPrice, IGasPriceProvider } from './gas-price-provider';
+
+export type BlockNativeGasPriceResponse = {
+  system: string;
+  network: string;
+  unit: string;
+  maxPrice: number;
+  currentBlockNumber: number;
+  msSinceLastBlock: number;
+  blockPrices: BlockPrice[];
+  estimatedBaseFees: { [block: string]: BaseFeeEstimate };
+};
+
+interface BlockPrice {
+  blockNumber: number;
+  estimatedTransactionCount: number;
+  baseFeePerGas: number;
+  estimatedPrices: PriceEstimate[];
+}
+
+interface PriceEstimate {
+  confidence: number;
+  price: number;
+  maxPriorityFeePerGas: number;
+  maxFeePerGas: number;
+}
+
+interface BaseFeeEstimate {
+  confidence: number;
+  baseFee: number;
+}
+
+export class BlockNativeGasPriceProvider extends IGasPriceProvider {
+  private apiKey: string;
+  private url: string;
+  constructor(apiKey: string) {
+    super();
+    this.apiKey = apiKey;
+    this.url = 'https://api.blocknative.com/gasprices/blockprices';
```

Comment:
Env var?

Reply:
We discussed it with papasmurf and decided against as the extra burden is not worth it